Lib/tarfile.py

```diff
@@ -2349,6 +2350,8 @@ def next(self):
                     raise ReadError(str(e)) from None
             except SubsequentHeaderError as e:
                 raise ReadError(str(e)) from None
+            except zlib.error as e:
+                raise ExtractError('archive may be corrupted or invalid') from e
```

Comment:
I don't think that the zlib traceback is useful.

```suggestion
                raise ReadError(f'zlib error: {e}')
```

Reply:
This should match the other branches:

    raise ReadError(str(e)) from None

Reply:
@jdevries3133, Ethan is right. Please make the ReadError in your new code also `from None`.

Reply:
Ok, note that the snippets in this thread are outdated; all of the changes requested here have been implemented.